### ISSUE TITLE
[backend] OutputArtifact path is None

### ISSUE THREAD

**Issue by wilbry** [kubeflow/pipelines#5560]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
Deployed Standalone on k3s
*  KFP version: 
1.50
*  KFP SDK version: 
1.5.0rc3

### Steps to reproduce

Create lightweight component using OutputArtifact, compiled with V1 compiler, using V2 Compatible, and run through the UI. The following error appears, I think in the effort to create the output artifact data library.

```bash
Traceback (most recent call last):
  File "/tmp/tmp.VtSD5WzDcx", line 594, in <module>
    executor_main()
  File "/tmp/tmp.VtSD5WzDcx", line 588, in executor_main
    function_to_execute=function_to_execute)
  File "/tmp/tmp.VtSD5WzDcx", line 353, in __init__
    artifacts_list[0])
  File "/tmp/tmp.VtSD5WzDcx", line 366, in _make_output_artifact
    return OutputArtifact(artifact_type=type(artifact), artifact=artifact)
  File "/tmp/tmp.VtSD5WzDcx", line 296, in __init__
    os.makedirs(self.path, exist_ok=True)
  File "/usr/local/lib/python3.7/os.py", line 208, in makedirs
    head, tail = path.split(name)
  File "/usr/local/lib/python3.7/posixpath.py", line 107, in split
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
F0428 16:41:59.832704      37 main.go:56] Failed to successfuly execute component: exit status 1
goroutine 1 [running]:
github.com/golang/glog.stacks(0xc000c80300, 0xc000652000, 0x61, 0xb6)
	/go/pkg/mod/github.com/golang/glog@v0.0.0-20160126235308-23def4e6c14b/glog.go:769 +0xb9
github.com/golang/glog.(*loggingT).output(0x2890520, 0xc000000003, 0xc0000fe000, 0x27d6a2a, 0x7, 0x38, 0x0)
	/go/pkg/mod/github.com/golang/glog@v0.0.0-20160126235308-23def4e6c14b/glog.go:720 +0x3b3
github.com/golang/glog.(*loggingT).printf(0x2890520, 0x3, 0x1b2c9c8, 0x2b, 0xc000a65f58, 0x1, 0x1)
	/go/pkg/mod/github.com/golang/glog@v0.0.0-20160126235308-23def4e6c14b/glog.go:655 +0x153
github.com/golang/glog.Fatalf(...)
	/go/pkg/mod/github.com/golang/glog@v0.0.0-20160126235308-23def4e6c14b/glog.go:1148
main.main()
	/build/cmd/launch/main.go:56 +0x357
```

### Expected result

The artifact should be created without error, so that the pipeline can run successfully. 

### Materials and Reference

I am working on a sample case in response to  the discussion taking place in #5453. The code at the moment is 

```python
from kfp import components
from kfp.dsl.io_types import OutputArtifact, Model
from kfp.dsl import pipeline, PipelineExecutionMode
from kfp.compiler import Compiler

def train_model(model: OutputArtifact(Model)) -> str:
  from sklearn.datasets import make_classification
  from sklearn.ensemble import RandomForestClassifier
  from joblib import dump
  import os

  fake_features, fake_labels = make_classification(n_features = 5)

  classifier = RandomForestClassifier(n_estimators=3)
  classifier.fit(fake_features, fake_labels)

  dump(classifier, os.path.join(model.path, 'model.joblib'))

  return model.uri

train_model_op = components.create_component_from_func_v2(train_model, packages_to_install=['sklearn'])
kf_serving_op = components.load_component_from_file('./components/kfserving_v2.yaml')

@pipeline(
  name='output_artifact_sample',
  description='Demonstrate using OutputArtifact URI information in next component',
  pipeline_root='minio://sample_output_artifiact_root'
)
def simple_kf_serving_pipeline():
    train_step = train_model_op()

    kf_serving_op(
            action = 'apply',
            model_name = 'output_artifact_sample',
            model_uri = train_step.outputs['Output'],
            framework = 'sklearn',
            service_account= 'sa',
            namespace='default')



def main():
    Compiler(mode = PipelineExecutionMode.V2_COMPATIBLE).compile(simple_kf_serving_pipeline, 'sample_artifact_pipeline.yaml')

if __name__ == "__main__":
    main()

```
---

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by wilbry** [comment]
The packaged KFServing component isn't currently compatible with V2, due to the Bool type no longer being supported as far as I know. The issue I got happens when that is fixed.

**Comment by Bobgy** [comment]
Hi @wilbry, thank you for your efforts porting kfserving component to use v2 semantics!
Sorry it took a while until I have time to investigate.

Your initial post uses OutputArtifact, but it doesn't exist. Please refer to our updated documentation: https://www.kubeflow.org/docs/components/pipelines/sdk/v2/python-function-components/

**Comment by Bobgy** [comment]
OutputArtifact(Model) should be changed to Output[Model]

@wilbry can you share your kfserving_v2.yaml component? I fixed your error, but cannot proceed, because I'm not sure what changes you made to kfserving_v2.yaml.

**Comment by Bobgy** [comment]
Quite a few fixes have been released to v2 compatible mode released in kfp sdk 1.7.0 and KFP backend 1.7.0-rc.3 is out. It'll be super helpful if you can have a try on the latest releases.
